jax/_src/ffi.py

```diff
@@ -15,9 +15,48 @@
 from __future__ import annotations
 
 import os
+import ctypes
 
 from jax._src.lib import jaxlib
 
+# Use Python stable C API function to construct a PyCapsule.
+# From ctypes.pythonapi documentation:
+#
+#  ctypes.pythonapi
+#    An instance of PyDLL that exposes Python C API functions as attributes.
+#    Note that all these functions are assumed to return C int, which is of
+#    course not always the truth, so you have to assign the correct restype
+#    attribute to use these functions.
+#
+# Taken from: https://docs.python.org/3/library/ctypes.html.
+#
+# More context here: https://stackoverflow.com/questions/24377845/ctype-why-specify-argtypes.
+#
+# Following this advice we annotate argument and return types of PyCapsule_New
+# before we call it. Example here:
+# https://stackoverflow.com/questions/65056619/converting-ctypes-c-void-p-to-pycapsule
+def build_capsule(funcptr):
+  """Construct a PyCapsule out of a ctypes function pointer.
+
+  A typical use for this is registering custom call targets with XLA:
+
+    import ctypes
+    import jax
+    from jax.lib import xla_client
+
+    fooso = ctypes.cdll.LoadLibrary('./foo.so')
+    xla_client.register_custom_call_target(
+      name="bar",
```

Comment:
Nit: Google Python style uses 4 space hanging indent.

Reply:
Fixed, thank you.